### DISCUSSION TITLE
Constant - "Internal error. Looks like something went wrong on our end" from SSG calls with NextJS

### DISCUSSION THREAD

**Discussion by joelpierre** [Shopify/storefront-api-feedback#184] [Help]
Regularly getting "Internal error. Looks like something went wrong on our end". Sometimes it's fine other times its really bad. I assume it's rate limiting? But I would expect a 429 status code or at least something of the sort in the response `errors` node.

Request ID: `4ffff949-eaa6-46ed-8d6e-77563f21d78c`

```
request: {
    query: 'query ProductsPaths($country: CountryCode!) @inContext(country: $country) {\n' +
      '  products(first: 10) {\n' +
      '    nodes {\n' +
      '      handle\n' +
      '    }\n' +
      '  }\n' +
      '}',
    variables: { country: 'GB' }
  }
```

**Comment by duncan-fairley** [comment]
What version of the storefront API are you using? Does the query work when `inContext` filter is removed?

**Reply by joelpierre** [reply]
> What version of the storefront API are you using? Does the query work when `inContext` filter is removed?

API Version: 2023-01

Haven't tried with `inContext` removed however we need this for our internationalised site. I can remove temporarily to test.

Was there anything from that request ID that could help?

**Reply by joelpierre** [reply]
Key thing too is it never happens in the GQL playground or Post man etc. Pretty much always happens during the build process for nextjs

**Reply by duncan-fairley** [reply]
I'm not with Shopify so can't check the request ID but have been working through similar issues with queries with context filters. Interesting that you can't replicate in Postman, I'd be able to replicate reliably in my codebase and within Postman.  For us it was related to one of the products being returned in the query being in a draft state.

**Reply by joelpierre** [reply]
Maybe I don't notice it in playground or postman due to the sheer amount of API calls SSG with nextjs does.

Hmm that is interesting... I will only do published products during build. For some reason I thought that draft products simply were not returned 🤔
